### PR TITLE
[PyTorch FE] instance_norm: honor use_input_stats

### LINKED ISSUE #37954: [Bug]: PyTorch frontend ignores `use_input_stats` of `instance_norm` — running statistics are used whenever provided, even when `use_input_stats=True`
## OpenVINO Version

2026.1.0 (build 2026.1.0-21367, releases/2026/1); also present on master (`src/frontends/pytorch/src/op/instance_norm.cpp` — `translate_instance_norm` dispatches on whether inputs 3/4 (`running_mean`/`running_var`) are `None` and never reads the `use_input_stats` argument).

## Operating System

Ubuntu 20.04.6 LTS (Focal Fossa), Linux x86_64 (kernel 5.15.0-139-generic)

## Device used for inference

CPU

## Framework

PyTorch (2.10.0+cu128), converted in-memory via `ov.convert_model` (PyTorch frontend)

## Model used

Minimal `nn.Module` calling `F.instance_norm(x, running_mean, running_var, weight, bias, use_input_stats=True)` on a `(2,3,4,4)` `float32` tensor.

## Issue description

`torch.nn.functional.instance_norm(input, running_mean, running_var, weight, bias, use_input_stats=True, ...)` documents: when `use_input_stats=True`, **input statistics** are used for normalization; the supplied `running_mean`/`running_var` are ignored (they only serve inference when `use_input_stats=False`). The PyTorch frontend converter dispatches instead purely on whether `running_mean`/`running_var` are provided:

```cpp
if (context.input_is_none(3) && context.input_is_none(4)) {
    return translate_instance_norm_inference(...);   // per-instance MVN over spatial dims (input statistics)
}
return translate_instance_norm_train(...);           // reshaped BatchNorm using the given running stats
```

So whenever `running_mean`/`running_var` are non-`None`, it always normalizes with those **running statistics**, and a call with `use_input_stats=True` + running statistics yields numerically wrong output (the `use_input_stats` flag — `context.input` 5 — is never consulted).

Measured with the reproducer below (`x` shape `(2,3,4,4)`, `eps=1e-5`):

| reference | maxdiff vs torch | maxdiff vs OpenVINO |
|---|---|---|
| per-(n,c) input statistics (torch's documented behavior for `use_input_stats=True`) | 2.4e-7 | ~2.26 |
| running-statistics normalization (what OpenVINO actually computes) | ~2.26 | 2.4e-7 |

OpenVINO output follows the running-statistics normalization and differs from torch by ~2.26 on this random input. Both the "running stats provided → use them" path and the "`running_mean`/`running_var` are `None` → use input statistics" path behave identically in torch and OpenVINO; only the combination `use_input_stats=True` **with** running statistics provided is mishandled (exactly the case the fuzzer and this repro exercise).

## Step-by-step reproduction

```python
#!/usr/bin/env python3
import warnings; warnings.filterwarnings("ignore")
import numpy as np, torch, torch.nn as nn, torch.nn.functional as F
import openvino as ov

class M(nn.Module):
    def forward(self, x, rm, rv, w, b):
        return F.instance_norm(x, rm, rv, w, b, use_input_stats=True)

torch.manual_seed(0)
x = torch.randn(2, 3, 4, 4); rm = torch.randn(3); rv = torch.rand(3) + 0.5
w = torch.randn(3); b = torch.randn(3)
m = M().eval()
with torch.no_grad():
    torch_out = m(x, rm, rv, w, b)

# torch's documented semantics for use_input_stats=True: normalize with input stats
eps = 1e-5
xs = x.view(2, 3, -1)
mean = xs.mean(-1, keepdim=True); var = xs.var(-1, unbiased=False, keepdim=True)
inst_ref = ((xs - mean) / torch.sqrt(var + eps)) * w.view(1, 3, 1) + b.view(1, 3, 1)
inst_ref = inst_ref.view_as(x)
print("torch vs input-stats ref maxdiff:", (torch_out - inst_ref).abs().max().item())  # ~2e-7 (correct)

ov_model = ov.convert_model(m, input=[(tuple(t.shape), t.dtype) for t in [x, rm, rv, w, b]])
ov_out = ov.Core().compile_model(ov_model, "CPU")([t.numpy() for t in [x, rm, rv, w, b]])
ov_out = np.asarray(list(ov_out.values())[0])
print("torch head:", torch_out.flatten()[:6].numpy())   # [-0.137 -0.114 -0.876 -0.721 -1.805 -1.672]
print("OV    head:", ov_out.flatten()[:6])              # [ 1.981  2.007  1.111  1.293  0.018  0.174]  <- wrong
print("OV vs torch maxdiff:", np.abs(ov_out - torch_out.numpy()).max())  # ~2.26
```

Expected: with `use_input_stats=True`, running statistics are ignored and the output is computed from input statistics (torch behavior above).
Actual: the converter normalizes with the supplied running statistics, giving a numerically wrong output (maxdiff ~2.26 on this input).

### PR DESCRIPTION
**Pull request by siyiweigeHEW**
### Details:
 - `instance_norm` with `use_input_stats=True` while running-mean/var are also provided used the running statistics instead of per-instance statistics.
 - Dispatch on use_input_stats: true -> per-instance normalization (MVN+affine, ignore running stats); false with stats present -> running statistics via reshaped BatchNormInference.
 - Adds layer tests for the use_input_stats=True + stats corner and a regression guard for use_input_stats=False + stats.

### Tickets:
 - Fixes #37954

### AI Assistance:
 - AI assistance used: yes
 - If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).
   AI assisted with root-cause analysis, the implementation and the regression tests. Human validation
   performed: a full local build of the PyTorch front-end; the layer tests for this op pass on the CPU reference device and in torch-export (FX) mode.

### CODE REVIEW

**Comment by siyiweigeHEW** [comment]
Hi @mvafin @bumbosiepsak @rkazants @CuriousPanCake, could you take a look at this PyTorch front-end fix when you have a moment?

In short: instance_norm with use_input_stats=True while running-mean/var were also provided used the running statistics instead of the per-instance ones. The change dispatches on use_input_stats — true gives per-instance normalization (MVN + affine, running stats ignored), false with stats present keeps the reshaped BatchNormInference — and adds layer tests for both corners.

The workflow runs for this PR are still waiting for approval, so nothing has run yet. Thanks!
